tools/signingtool/src/test/java/com/horizen/SigningToolTest.java

```diff
@@ -0,0 +1,224 @@
+        argsJson.put("prefix", prefix);
+        argsJson.put("type", SCHNORR);
+        args = argsJson.toString();
+        result = runTest(new String[]{VALIDATE_MESSAGE, args});
+        assertTrue(result != null && result.contains("valid") && result.contains("false"));
+
+        //Invalid signature
+        argsJson = new ObjectMapper().createObjectNode();
+        argsJson.put("message", testMessage);
+        argsJson.put("signature", signature + "123");
+        argsJson.put("publicKey", BytesUtils.toHexString(PROPOSITION_SERIALIZER.toBytes(publicKey)));
+        argsJson.put("prefix", prefix);
+        argsJson.put("type", SCHNORR);
+        args = argsJson.toString();
+        result = runTest(new String[]{VALIDATE_MESSAGE, args});
+        assertTrue(result != null && result.contains("error") && result.contains("Unable to parse signature"));
+
+        //Invalid publicKey
+        argsJson = new ObjectMapper().createObjectNode();
+        argsJson.put("message", testMessage);
+        argsJson.put("signature", signature);
+        argsJson.put("publicKey", BytesUtils.toHexString(PROPOSITION_SERIALIZER.toBytes(publicKey)) + "123");
+        argsJson.put("prefix", prefix);
+        argsJson.put("type", SCHNORR);
+        args = argsJson.toString();
+        result = runTest(new String[]{VALIDATE_MESSAGE, args});
+        assertTrue(result != null && result.contains("error") && result.contains("Unable to parse publicKey"));
+    }
+
+    private SchnorrSecret generateSecret() {
+        Random rnd = new Random();
+        byte[] vrfSeed = new byte[32];
+        rnd.nextBytes(vrfSeed);
+        return SchnorrKeyGenerator.getInstance().generateSecret(vrfSeed);
+    }
+
+    @Test
+    public void testPrivKeyToPubkey() throws IOException {
+        SchnorrSecret secretKey = generateSecret();
+
+        ObjectNode argsJson = new ObjectMapper().createObjectNode();
+        argsJson.put("privateKey", BytesUtils.toHexString(SECRETS_COMPANION.toBytes(secretKey)));
+        argsJson.put("type", SCHNORR);
+        String args = argsJson.toString();
+
+        String result = runTest(new String[]{PRIV_KEY_TO_PUB_KEY, args});
+        assertTrue(result != null && result.contains("publicKey") && !result.contains("error"));
+
+        JsonNode signatureJson = new ObjectMapper().readTree(result);
+        String publicKey = signatureJson.get("publicKey").asText();
+
+        assertEquals(BytesUtils.toHexString(PROPOSITION_SERIALIZER.toBytes(secretKey.publicImage())), publicKey);
+
+        //Invalid privateKey
+        argsJson = new ObjectMapper().createObjectNode();
+        argsJson.put("privateKey", BytesUtils.toHexString(SECRETS_COMPANION.toBytes(secretKey)) + "123");
+        argsJson.put("type", SCHNORR);
+        args = argsJson.toString();
+
+        result = runTest(new String[]{PRIV_KEY_TO_PUB_KEY, args});
+        assertTrue(result != null && result.contains("error") && result.contains("Unable to parse privateKey"));
+    }
+
+}
```

Comment:
@i-skrypnyk when running inside a container without networking (like in the secure enclave):
```
java -jar tools/signingtool/target/sidechains-sdk-signingtools-0.5.0-SNAPSHOT.jar
2022-10-28 10:34:11,405 main ERROR Could not determine local host name java.net.UnknownHostException: 95c76d23dcac: 95c76d23dcac: Temporary failure in name resolution
        at java.base/java.net.InetAddress.getLocalHost(Unknown Source)
        at org.apache.logging.log4j.core.util.NetUtils.getLocalHostname(NetUtils.java:54)
        at org.apache.logging.log4j.core.LoggerContext.lambda$setConfiguration$0(LoggerContext.java:620)
        at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(Unknown Source)
        at org.apache.logging.log4j.core.LoggerContext.setConfiguration(LoggerContext.java:620)
        at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:699)
        at org.apache.logging.log4j.core.LoggerContext.reconfigure(LoggerContext.java:716)
        at org.apache.logging.log4j.core.LoggerContext.start(LoggerContext.java:270)
        at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:155)
        at org.apache.logging.log4j.core.impl.Log4jContextFactory.getContext(Log4jContextFactory.java:47)
        at org.apache.logging.log4j.LogManager.getContext(LogManager.java:196)
        at org.apache.logging.log4j.LogManager.getLogger(LogManager.java:599)
        at com.horizen.SigningTool.main(SigningTool.java:27)
Caused by: java.net.UnknownHostException: 95c76d23dcac: Temporary failure in name resolution
        at java.base/java.net.Inet4AddressImpl.lookupAllHostAddr(Native Method)
        at java.base/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(Unknown Source)
        at java.base/java.net.InetAddress.getAddressesFromNameService(Unknown Source)
        at java.base/java.net.InetAddress$NameServiceAddresses.get(Unknown Source)
        at java.base/java.net.InetAddress.getAllByName0(Unknown Source)
        ... 13 more

{"Usage":["From command line: <program name> <command name> [<json data>]"],"Supported commands":["help","createSignature <arguments>","verifySignature <arguments>","signMessage <arguments>","validateMessage <arguments>","privKeyToPubKey <arguments>"]}
```

Reply:
@cronicc Hi, can you please check with the latest fix, I removed the logging completely for the signingtool, and in my tests log4j is no longer initialized, thus not invoking InetAddress.getLocalHost. But I don't know how to correctly emulate your environment.

Reply:
@i-skrypnyk error is gone now. To replicate `docker run --rm -v $(pwd)/tools/signingtool/target/:/target --network=none eclipse-temurin:11-jre-jammy java -jar /target/sidechains-sdk-signingtools-0.5.0-SNAPSHOT.jar`